pandas/core/internals/concat.py

```diff
@@ -65,13 +66,15 @@ def concatenate_block_managers(
             blk = join_units[0].block
             vals = [ju.block.values for ju in join_units]
 
-            if not blk.is_extension or blk.is_datetimetz or blk.is_categorical:
+            if not blk.is_extension or blk.is_datetimetz:
                 # datetimetz and categorical can have the same type but multiple
```

Comment:
the comment here may need to be updated

Reply:
Removed the comment (and the datetimetz special case) entirely now